### ISSUE TITLE
Upgrade PHP (move away from v7)

### ISSUE THREAD

**Issue by binarykitchen** [mail-in-a-box/mailinabox#2068]
Currently mail-in-a-box still uses PHP v7.2 but it's reached EOL and some Nextcloud apps do not run anymore on that version.

Thanks!

**Comment by binarykitchen** [comment]
Also wonder if I'm safe to bump PHP via command line separately?

**Comment by myfirstnameispaul** [comment]
Mail-in-a-Box uses the Ubuntu main repository version of PHP, which will receive security updates until Ubuntu 18.04 is EOL.

Mail-in-a-Box currently uses Nextcloud 20, which supports PHP 7.2.

**Comment by binarykitchen** [comment]
Understand but what can I do?

Should I wait until a newer mail-in-a-box version is out, or should I upgrade Ubuntu and PHP myself?

**Comment by myfirstnameispaul** [comment]
You should update Mail-in-a-Box as the project updates it.

If there is something you want to do with a server other than what Mail-in-a-Box already does, best to do it on a different server.

**Comment by binarykitchen** [comment]
Well, I'm already on latest version v55, yet it installs older PHP version. I'm on Ubuntu v18 - should I upgrade the Ubuntu version?

**Comment by myfirstnameispaul** [comment]
No. You have everything correctly installed. The version of PHP is the version currently supported by the Mail-in-a-Box project. If you upgrade the Ubuntu version, you will break Mail-in-a-Box.

**Comment by binarykitchen** [comment]
Ok, good. Are any upcoming updates planned soon regarding this issue?

**Comment by myfirstnameispaul** [comment]
From the perspective of the project, this is not currently an issue. Updates are release as-needed by the project maintainer, usually once a month or so.

**Comment by binarykitchen** [comment]
Fair enough but some Nextcloud apps require newer PHP versions and it will break soon ...

**Comment by myfirstnameispaul** [comment]
If these apps are not installed by Mail-in-a-Box, then it will not be an issue for the project.

**Comment by stvhay** [comment]
NC 20 only supports PHP 7. That said, the NC instance is recommending upgrade to 7.4.

**Comment by JoshData** [comment]
I think this issue is resolved --- there won't be any updates to PHP in Mail-in-a-Box until the next major update to Mail-in-a-Box is done for supporting Ubuntu 22.04. See #2083.
